delocate/tests/test_scripts.py

```diff
@@ -643,18 +653,186 @@ def test_glob(
+    script_runner.run(
+        [
+            "delocate-wheel",
+            whl_10_9,
+            "--require-target-macos-version",
+            "10.9",
+            "--ignore-missing-dependencies",
+        ],
+        check=True,
+        cwd=tmp_path,
+    )
+
+
+@pytest.mark.xfail(  # type: ignore[misc]
+    sys.platform != "darwin", reason="Needs macOS linkage."
+)
+def test_delocate_wheel_missing_architecture(
+    plat_wheel: PlatWheel,
+    script_runner: ScriptRunner,
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    shutil.copy(
+        plat_wheel.whl,
+        tmp_path / "plat2-1.0-cp311-cp311-macosx_10_9_universal.whl",
+    )
+    result = script_runner.run(
+        [
+            "delocate-wheel",
+            tmp_path / "plat2-1.0-cp311-cp311-macosx_10_9_universal.whl",
+        ],
+        check=False,
+        cwd=tmp_path,
+    )
+    assert result.returncode != 0
+    assert (
+        "Failed to find any binary with the required architecture: 'i386'"
+        in result.stderr
+    )
+
+
+@pytest.mark.xfail(  # type: ignore[misc]
+    sys.platform != "darwin", reason="Needs macOS linkage."
+)
+def test_delocate_wheel_verify_name_universal2_verify_crash(
+    plat_wheel: PlatWheel,
+    script_runner: ScriptRunner,
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    zip2dir(plat_wheel.whl, tmp_path / "plat")
+    shutil.copy(
+        DATA_PATH / "libam1_12.dylib",
+        tmp_path / "plat" / "fakepkg1" / "libam1.dylib",
+    )
+    whl_10_9 = tmp_path / "plat2-1.0-cp311-cp311-macosx_10_9_universal2.whl"
+    dir2zip(tmp_path / "plat", whl_10_9)
+    result = script_runner.run(
+        ["delocate-wheel", whl_10_9, "--require-target-macos-version", "10.9"],
+        check=False,
+        cwd=tmp_path,
+    )
+    assert result.returncode != 0
+    assert "Library dependencies do not satisfy target MacOS" in result.stderr
+    assert "libam1.dylib has a minimum target of 12.0" in result.stderr
+
+
+@pytest.mark.xfail(  # type: ignore[misc]
+    sys.platform != "darwin", reason="Needs macOS linkage."
+)
+@pytest.mark.script_launch_mode("subprocess")
+def test_delocate_wheel_verify_name_universal2_verify_crash_env_var(
+    plat_wheel: PlatWheel,
+    script_runner: ScriptRunner,
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    zip2dir(plat_wheel.whl, tmp_path / "plat")
+    shutil.copy(
+        DATA_PATH / "libam1_12.dylib",
+        tmp_path / "plat" / "fakepkg1" / "libam1.dylib",
+    )
+    whl_10_9 = tmp_path / "plat2-1.0-cp311-cp311-macosx_10_9_universal2.whl"
+    dir2zip(tmp_path / "plat", whl_10_9)
+    delocate_wheel_path = pjoin(
+        os.path.dirname(sys.executable), "delocate-wheel"
+    )
+
+    result = script_runner.run(
+        [
+            delocate_wheel_path,
+            whl_10_9,
+        ],
+        check=False,
+        cwd=tmp_path,
+        env={"MACOSX_DEPLOYMENT_TARGET": "10.9"},
+    )
```

Comment:
I do not understand what caused this problem. I jut hard hit into this.

Reply:
I don't like this workaround since it implies an incomplete installation of Delocate's scripts for testing, or an unexpected Python environment. I won't accept this workaround without a proper explanation of why the normal code doesn't work.

If you can't explain it then read `MACOSX_DEPLOYMENT_TARGET` at run-time instead of import-time so that this can run like any other test and then the workaround will no longer be needed. Otherwise, at least show me the error or traceback.

Reply:
My mistake, I've been reading outdated commits and missed that these issues are already resolved.

Reply:
Error is              
```
FileNotFoundError: [Errno 2] No such file or directory: 'delocate-wheel'
```

It looks like PATH is not properly propagated.